### PR TITLE
Add `SharedKeys::disableCaching`

### PR DESCRIPTION
**Pull request by blaugold**
@snej Thanks for the advice in https://github.com/couchbase/couchbase-lite-core/pull/1642#issuecomment-1536548762. This 
 PR is implementing the suggested changes.

### CODE REVIEW

File: Fleece/Core/SharedKeys.hh [hunk 1]
```diff
@@ -143,6 +149,8 @@ namespace fleece { namespace impl {
         mutable std::mutex _mutex;
         unsigned _count {0};
         bool _inTransaction {true};                     // (for PersistentSharedKeys)
+        bool _isCachingEnabled {true};
+        bool _isCacheable {true};
```
Comment (jianminzhao) [inline Fleece/Core/SharedKeys.hh:153]: Why do you need two of the above?

File: Tests/SharedKeysTests.cc [hunk 2]
```diff
@@ -97,6 +97,8 @@ TEST_CASE("decode", "[SharedKeys]") {
 
 TEST_CASE("revertToCount", "[SharedKeys]") {
     Retained<SharedKeys> sk = new SharedKeys();
+    sk->disableCaching();
```
Comment (jianminzhao) [inline Tests/SharedKeysTests.cc:100]: Whereas we want to ensure diableCaching does not affect the following checks, the other aspect is ensure get(Dict::key &keyToFind) works as expected.

**Review by jianminzhao — CHANGES_REQUESTED** [review]

**Comment by jianminzhao** [comment]
@blaugold are you still interested in moving forward? If not, I am going to close it.

**Comment by blaugold** [comment]
@jianminzhao Thanks for reviewing this PR! I still want to work on this, I just haven't had time to respond yet. I hope I can get to it this week.

**Pull request closed without merging**